### PR TITLE
Support `disk.driver` hw requirement for `mrack`

### LINKED ISSUE #2546: how to filter server with boot disk driver
Install OS to special host disk, like megaraid_sas, mpt3sas
so request tmt add filter server with BOOTDISK.

please refer the beaker job xml

<key_value key="BOOTDISK" op="=" value="megaraid_sas"/>
<key_value key="BOOTDISK" op="=" value="mpt3sas"/>

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] update the specification
* [x] include a release note

Resolves #2546

### CODE REVIEW

File: tmt/steps/provision/mrack.py [hunk 1]
```diff
@@ -144,6 +145,20 @@ def __init__(self, name: str, operator: str, value: str) -> None:
             }
 
 
+@dataclasses.dataclass(init=False)
+class MrackHWKV(MrackHWElement):
```
Comment (happz) [inline tmt/steps/provision/mrack.py:149]: `MrackHWKeyValue` would look better, `HWKV` is hard to grasp at quick glance.

**Review by happz — APPROVED** [review]

File: tmt/steps/provision/mrack.py [hunk 2]
```diff
@@ -145,6 +146,20 @@ def __init__(self, name: str, operator: str, value: str) -> None:
             }
 
 
+@dataclasses.dataclass(init=False)
+class MrackHWKeyValue(MrackHWElement):
+    """ An key-value element """
```
Comment (psss) [inline tmt/steps/provision/mrack.py:151]:
```suggestion
    """ A key-value element """
```

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for implementing this. Just one nitpick. Could you please also update the release note accordingly? The checklist in the pull request description could remind about this ;-) We remove irrelevant items from it and mark as done completed items.

**Comment by skycastlelily** [comment]
Updated:)

> The checklist in the pull request description could remind about this ;-)

Thanks for your reminder,will keep that in mind:)

**Review by psss — APPROVED** [review]

**Pull request merged**